bundles/org.openhab.binding.goveelan/src/main/java/org/openhab/binding/goveelan/internal/GoveeLanHandler.java

```diff
@@ -0,0 +1,369 @@
+            LOGGER.error("exception when receiving status packet {}", Arrays.toString(e.getStackTrace()));
+            // as we haven't received a packet we also don't know where it should have come from
+            // hence, we don't know which thing put offline.
+            // a way to monitor this would be to keep track in a list, which device answers we expect
+            // and supervise an expected answer within a given time but that will make the whole
+            // mechanism much more complicated and may be added in the future
+        } finally {
+            refreshJobRunning = false;
+        }
+    };
+
+    /**
+     * This thing related job <i>thingRefreshSender</i> triggers an update to the Govee device.
+     * The device sends it back to the common port and the response is
+     * then received by the common #refreshStatusReceiver
+     */
+    private final Runnable thingRefreshSender = () -> {
+        try {
+            triggerDeviceStatusRefresh();
+            if (!thing.getStatus().equals(ThingStatus.ONLINE)) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@text/error.could-not-query-device [\"" + goveeLanConfiguration.hostname + "\"]");
+        }
+    };
+
+    public GoveeLanHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        goveeLanConfiguration = getConfigAs(GoveeLanConfiguration.class);
+
+        final String ipAddress = goveeLanConfiguration.hostname;
+        if (!ipAddress.isEmpty()) {
+            THING_HANDLERS.put(ipAddress, this);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/error.offline.ip-address.missing");
+            return;
+        }
+        if (!THING_HANDLERS.isEmpty()) {
+            startRefreshStatusJob();
+        }
+
+        if (triggerStatusJob == null) {
+            LOGGER.debug("Starting refresh trigger job for thing {} ", thing.getLabel());
+
+            triggerStatusJob = scheduler.scheduleWithFixedDelay(thingRefreshSender, 100,
+                    goveeLanConfiguration.refreshInterval * 1000L, TimeUnit.MILLISECONDS);
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+    }
+
+    /**
+     * Stop the Refresh Status Job, so the same socket can be used for something else (like discovery)
+     */
+    public static void stopRefreshStatusJob() {
+        if (refreshStatusJob != null) {
+            refreshStatusJob.cancel(true);
+        }
+        refreshStatusJob = null;
+        refreshJobRunning = false;
+    }
+
+    /**
+     * (re)start the refresh status job
+     */
+    public static void startRefreshStatusJob() {
+        if (refreshStatusJob == null) {
+            refreshStatusJob = ThreadPoolManager.getScheduledPool("goveeThingHandler")
```

Comment:
Is there any reason why you need your own thread pool?

Reply:
Nope, I think that was your recommendation above.
This is how it was before: 

> refreshStatusJob = ThreadPoolManager.getScheduledPool("thingHandler")

Reply:
Sorry, previously I only commented the naming, now I'm getting into the substance. Why a custom thread pool at all?

Reply:
The answer is simple: I looked how other bindings do it and afaik all initialize it by providing it its own name. Therefore I did it the same way.

This wouldn't work

> refreshStatusJob = ThreadPoolManager.getScheduledPool()

What is your proposal then? Use THREAD_POOL_NAME_COMMON 

Only few people do it that way but I am surely open to that.

Reply:
Since your handler is derived from `BaseThingHandler`, you can simply use `scheduler` which is set up to use a shared pool:
```java
refreshStatusJob = scheduler.scheduleWithFixedDelay(REFRESH_STATUS_RECEIVER, 100, 1000, TimeUnit.MILLISECONDS);
```

Reply:
Unfortunately it does work due to the scheduler not being static

<img width="1027" alt="image" src="https://github.com/openhab/openhab-addons/assets/5937600/6b2a8838-3387-4668-a384-1db52a399f9d">